perfkitbenchmarker/packages/ycsb.py

```diff
@@ -345,19 +350,26 @@ def CombineHistograms(hist1, hist2):
       result.append((k, h1.get(k, 0) + h2.get(k, 0)))
     return result
 
+  result = copy.deepcopy(result_list[0])
+  DropUnaggregated(result)
+
   for indiv in result_list[1:]:
     for group_name, group in indiv['groups'].iteritems():
+      if group_name not in result['groups']:
+        logging.warn('"%s" in new dict, but not in old.', group_name)
+        result['groups'][group_name] = copy.deepcopy(group)
+        continue
       for k, v in group['statistics'].iteritems():
-        if k not in result['groups'][group_name]['statistics']:
-          logging.warn('"%s" in new dict, but not in old.', k)
-          result['groups'][group_name]['statistics'][k] = copy.deepcopy(v)
-          continue
-        elif k not in operators:
+        if k not in operators:
```

Comment:
Consider adding some inline comments on the logic here on the flow. It looks a bit confusing why you move the code around.